clang/runtime/dpct-rt/include/dpl_extras/vector.h

```diff
@@ -66,143 +67,191 @@ class device_vector {
 
   void _set_capacity_and_alloc() {
     _capacity = ::std::max(_size * 2, _min_capacity());
-    _storage = _alloc.allocate(_capacity);
+    _storage = ::std::allocator_traits<Allocator>::allocate(_alloc, _capacity);
```

Comment:
Using an alias for `::std::allocator_traits<Allocator>::` may simplify the code.